prov/verbs/src/verbs_cq.c

```diff
@@ -230,47 +230,181 @@ static void fi_ibv_cq_read_data_entry(struct ibv_wc *wc, int i, void *buf)
 		wc->byte_len : 0;
 }
 
-static int fi_ibv_match_ep_id(struct slist_entry *entry, const void *ep_id)
+static inline int fi_ibv_match_ep_id(struct slist_entry *entry,
+				     const void *ep_id)
 {
-	struct fi_ibv_msg_epe *epe = container_of(entry, struct fi_ibv_msg_epe, entry);
+	struct fi_ibv_msg_epe *epe =
+		container_of(entry, struct fi_ibv_msg_epe, entry);
 
-	if (epe->ep->ep_id == (uint64_t) ep_id)
-		return 1;
+	return (epe->ep->ep_id == (uint64_t)ep_id);
+}
 
-	return 0;
+static inline int fi_ibv_poll_outstanding_cq(struct fi_ibv_msg_ep *ep,
```

Comment:
This function looks similar to fi_ibv_poll_cq. Can parts of it be extracted out to create a common function for both?

Reply:
It's a bit complicated to split the common parts. It was possible to split only part that is responsible to handle signaled events. Please, check what is done

Reply:
I have suggested some changes. Can you check if after making those changes the function becomes simplified enough that it can be merged with the other one / use a common function?